eclipse-jdt/eclipse.jdt.ui
# Folding mechanism for while/for/if/switch-case

Problem:
Currently, there is no folding mechanism for while/for/if/switch-case statements in Eclipse. VSCode and IntelliJ support a folding mechanism for these statements. Therefore, I have created a method that implements this folding mechanism. This is also an open issue: #1426.

Before:
![image](https://github.com/user-attachments/assets/396b4061-9254-4896-b16a-f245026a0119)

After:
![image](https://github.com/user-attachments/assets/406f91c0-de52-4707-a611-c8016cb31245)


How to test:
1. Start an Eclipse workspace and create a new class.
2. Add while/for/if/switch-case statements to this class.
3. Verify that the folding mechanism works as expected.

Here is a small program you can use to test:
public class SimpleProgram {
    public void checkNumber(int number) {
        if (number > 0) {
            System.out.println(number + " ist positiv.");
        } else {
            System.out.println(number + " ist null.");
        }
    }

    public static void main(String[] args) {
        SimpleProgram program = new SimpleProgram();
        int count = 1;
        while (count <= 5) {
            System.out.println("Count ist: " + count);
            count++;
        }
    }
}

Bugs:
- [ ] #1539
- [ ] When there are multiple nested while/for/if/switch-case statements, not all foldable regions are recognized.
